src/KeyManager.sol

```diff
@@ -71,24 +68,54 @@ abstract contract KeyManagerBase {
 
 contract KeyManager_v0 is KeyManagerBase {
     IAndromeda public Suave;
+    BIP32 public bip32;
+    // placeholder for hardened and non_hardened child indexes, start index is 2^31
+    uint32 public constant HARDENED_START_INDEX = 0x80000000;
 
     constructor(address _Suave) {
         Suave = IAndromeda(_Suave);
+        bip32 = new BIP32(Suave);
+    }
+
+    function hardenAddress(address a) internal override pure returns (uint32) {
+        return uint32(uint256(keccak256(abi.encode(a))) | HARDENED_START_INDEX);
+    }
+
+    // Any contract in confidential mode can request a
+    // hardened derived key
+    function _derivedPriv(address a) internal override returns (bytes32) {
+        bytes32 _seed = seed();
+        uint32 addrIndex = hardenAddress(a);
+        (BIP32.ExtendedPrivateKey memory _xPriv, BIP32.ExtendedPublicKey memory _xPub) = bip32.deriveChildKeyPairFromSeed(abi.encodePacked(_seed), addrIndex);
+        return _xPriv.key;
+    }
+
+    function derivedPriv() public returns (bytes32) {
+        return _derivedPriv(msg.sender);
+    }
+
+    function seed() private returns (bytes32) {
```

Comment:
this function being called `seed` means all variables that youd normally call `seed` have to be called `_seed` - I'd suggest naming this function `getSeed` and the variables `seed`